src/v/cluster/rm_stm.cc

```diff
@@ -2782,6 +2803,9 @@ rm_stm::load_abort_snapshot(abort_index index) {
       meta_parser.bytes_left());
 
     auto data_buf = co_await read_iobuf_exactly(reader->input(), snapshot_size);
+    uint64_t _abort_snapshot_disk_size = co_await reader->get_snapshot_size();
+    _abort_snapshot_sizes.emplace(
+      std::make_pair(index.first, index.last), _abort_snapshot_disk_size);
```

Comment:
I think this will underestimate the size as abort snapshots are loaded lazily so not all sizes are going to be present in the map (in fact majority probably won't).

I am also curious how these abort snapshots will get re-generated after a partition moves (the size of a newly-created replica is what we are interested in the end).